### PR TITLE
[DOCS] Clarify rolling upgrade support for minor versions

### PR DESCRIPTION
**Pull request by jrodewig**
We only support rolling upgrades between minor versions of the same
major.

It also adds some tags to reuse duplicated content.

### Previews
* Upgrade Elasticsearch: https://elasticsearch_81444.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/setup-upgrade.html
* Rolling upgrades: https://elasticsearch_81444.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/rolling-upgrades.html

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachince update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/upgrade.asciidoc [hunk 1]
```diff
@@ -16,13 +16,15 @@ endif::[]
 {es} can usually be upgraded using a <<rolling-upgrades,Rolling upgrade>>
 process so upgrading does not interrupt service. Rolling upgrades are supported:
 
-* Between minor versions
+// tag::rolling-upgrade-versions[]
+* Between minor versions of the same major version
```
Comment (jrodewig) [inline docs/reference/upgrade.asciidoc:20]: @debadair You mind taking a look at this. It's a pretty minor change, but it's important for clarity.

**Review by mark-vieira — APPROVED** [review]
Change LGTM

**Comment by jrodewig** [comment]
Thanks @mark-vieira!

**Pull request merged**